### DISCUSSION TITLE
What is the suggest way of integrating with OAuth2 / OpenID Connected based systems

### DISCUSSION THREAD

**Discussion by IanPattersonMuo** [AvaloniaUI/Avalonia#8407] [Q&A]
What is the suggested method of integrating with a service like Auth0 / OpenID Connect from Avalonia apps ?

**Comment by maxkatz6** [comment]
Normally Auth0 / OpenID libraries like MSAL support custom implementation of native webview that asks user to show the login page. You can provide this implementation using any of available WebView controls for Avalonia.

Alternatively, some of these libraries also provider ready to use native implementation that opens separated window. In this case it would just work. MSAL does it at least for Windows and macOS from what I remember.

**Reply by workgroupengineering** [reply]
Are there any plans to implement WebView without using the third party libraries?

**Comment by beto-rodriguez** [comment]
Is there any link available?

I am having a bad time to implement this

**Reply by ennerperez** [reply]
https://www.nuget.org/packages/Auth0Avalonia
